woocommerce/wc-admin
# Render psuedo wc pay payment gateway in payment method table

Partly addresses #7319 

Adds a psuedo WC Pay payment gateway to the payment method list. This is the main part of work that is required for #7319.
The changes are under a feature flag at the moment only for development.
Items that are done in a separate PR:
- Add copy from woocommerce.com
- Add the explat experiment
- Enable feature flag in all environments and add changelog

No changelog as it will be added in the final PR.

### Screenshots

<img width="1533" alt="Screen Shot 2021-08-24 at 8 45 11 AM" src="https://user-images.githubusercontent.com/2240960/130610888-bc2050ff-a343-4d42-8547-2a21e832e766.png">

### Detailed test instructions:

- Set the `tasks` config in development.json to `false` to make testing a bit easier ( run `npm run build:feature-config` after ) and make sure you don't have WooCommerce Payments installed or enabled.
- Go to **WooCommerce > Settings > Payments**
- WooCommerce Payments should appear in the payment method list (at the top) and not in the Payment recommendations card rendered below the table.
- Also check if the `wcadmin_settings_payments_recommendations_pageview` track is triggered in your console, you can have it shown up by using `localStorage.setItem( 'debug', 'wc-admin:*' );`)
    - The track should contain props for each plugin displayed `<plugin name>-displayed: true`
- Aside from the copy/description the method should look like this: https://github.com/woocommerce/woocommerce-admin/issues/7319#issuecomment-897066436 (with the enabled button hidden as well) (cc @elizaan36 the icons in the referenced comment are pushed right, but in the [figma](https://d.pr/i/ny4bbW) they are pushed left do you have a preference? )
- Try the sorting and click save changes, the location should persist 
- Click `Install`, the button should show as busy and once finished redirect to the WooCommerce Payments set up screen.
- Go to **WooCommerce > Settings > Payments** the psuedo gateway should not be showing anymore.
- Check if the page view track is still triggered having the `woocommerce-payments-displayed` set  to false.
- Go to **Plugins > Installed Plugins** and disable and or remove WooCommerce Payments again
- Go to **WooCommerce > Home > Set up payments** (WooCommerce Payments should not appear as already installed)
- Go to the free features tab in **WooCommerce > Home > Store details > Business details**, WooCommerce payments should be part of the free features list (in a supported country of-course).
